src/main/java/io/cdap/plugin/gcp/spanner/source/SpannerSource.java

```diff
@@ -272,6 +309,18 @@ private Schema getSchema(FailureCollector collector) {
 
   }
 
+  private Statement getStatementForOneRow(String importQuery) {
+    String query;
+    if (StringUtils.containsIgnoreCase(importQuery, LIMIT)) {
+      int total = StringUtils.lastIndexOf(importQuery, LIMIT);
+      String substringToReplace = importQuery.substring(total);
+      query = importQuery.replace(substringToReplace, "limit 1");
```

Comment:
This can break in some corner cases where the table name contains "limit" substring. For example something like `SELECT <columns> from limited`.
Can you make sure we only replace "limit < number >" with "limit 1".

Reply:
Done